kolibri/core/content/serializers.py

```diff
@@ -122,6 +124,52 @@ class Meta:
         )
 
 
+class PublicChannelSerializer(KolibriModelSerializer):
+    matching_tokens = serializers.SerializerMethodField('match_tokens')
+    language = serializers.SerializerMethodField()
+    included_languages = serializers.SerializerMethodField()
+    total_resource_count = serializers.SerializerMethodField()
+    icon_encoding = serializers.SerializerMethodField()
+    published_size = serializers.SerializerMethodField()
+    last_published = serializers.SerializerMethodField()
+    public = serializers.SerializerMethodField()
+
+    def get_public(self, instance):
+        return True
+
+    def get_language(self, instance):
+        if instance.root.lang is None:
+            return None
+
+        return instance.root.lang.lang_code
+
+    def get_included_languages(self, instance):
+        channel_nodes = ContentNode.objects.filter(channel_id=instance.id)
+        return list(channel_nodes.order_by('lang').values_list('lang', flat=True).distinct())
+
+    def get_total_resource_count(self, instance):
+        channel_nodes = ContentNode.objects.filter(channel_id=instance.id)
+        return channel_nodes.filter(available=True).exclude(kind=content_kinds.TOPIC).count()
+
+    def get_icon_encoding(self, instance):
+        return instance.thumbnail
+
+    def get_published_size(self, instance):
+        channel_nodes = ContentNode.objects.filter(channel_id=instance.id)
+        return _total_file_size(_files_for_nodes(channel_nodes).filter(available=True))
+
+    def get_last_published(self, instance):
+        return instance.last_updated if not instance.last_updated else create_timezonestamp(instance.last_updated)
+
+    def match_tokens(self, channel):
```

Comment:
need to add the token fields to kolibri models.

Reply:
Might not be necessary for peer-to-peer. Will every channel on a peer device have `public = True` by default? In that case, the channel will appear in the channel list, and the user will probably not need to enter a token.

Reply:
yeah so every channel will not have public set to `True`. Some content creators would like to lock down their content, hence the need for these tokens.